snoop/views.py

```diff
@@ -59,18 +67,26 @@ def _as_eml(doc):
     if doc.content_type == 'application/vnd.ms-outlook':
         return str(Path(doc.filename).with_suffix('.eml'))
 
-def document_as_eml(request, id):
-    doc = get_object_or_404(models.Document, id=id)
+def document_as_eml(request, collection_slug, id):
+    doc = get_object_or_404(
+        models.Document,
+        id=id,
+        collection__slug=collection_slug,
+    )
     if not _as_eml(doc):
         return HttpResponseNotFound()
     with open_msg(doc) as f:
         return HttpResponse(f.read(), content_type='message/rfc822')
 
-def _process_document(id):
+def _process_document(collection_slug, id):
     parent_id = None
     attachments = []
 
-    doc = get_object_or_404(models.Document, id=id)
+    doc = get_object_or_404(
```

Comment:
We're doing this `get_object_or_404` call in four places. How about a helper function?
```python
def find_doc(collection_slug, id):
    return get_object_or_404(
        models.Document,
        id=id,
        collection__slug=collection_slug,
    )
```